woocommerce/woocommerce-android
# Configuration immutable

Part of : #10499 

>[ Bug ] When accessing the Configuration screen for a bundle that’s already added to an order, tapping the X will save changes to the configuration, including letting a user save an invalid configuration (or, at least, it appears to – then in the order summary it shows a valid configuration).

### Description
As discovered on the bundles testing plan, changing the configuration of an already configured product without saving the changes could change (locally) the configuration. This happened because the memory reference of the children's configuration was the same for the item added to the order and the configuration sent to edit in the configuration screen. To fix this issue, I made the product configuration immutable, and now, if the children's configuration changes, it points to a memory reference different from the configuration in the order.

### Testing instructions
1. Open the orders tab
2. Tap the add order button (+)
3. Tap on add products
4. Select a bundled product that requires a configuration
5. Configure the bundled product and save the configuration
6. Tap on Select 1 product
7. Expand the recently added product
8. Tap on configure
9. Modify the configuration
10. Tap on the close button at the top of the screen (x)
11. Expand the product again
12. Tap on configure again
13. Check that the product maintains the same configuration as the one in the order being created and not the changes you applied before tapping the close button (x)

### Images/gif

#### Before

https://github.com/woocommerce/woocommerce-android/assets/18119390/f5cc3ac1-b634-4ddd-9623-ea4b86f8560a

#### After

https://github.com/woocommerce/woocommerce-android/assets/18119390/aea51f52-efcc-43ab-a894-70b94b907939
